erpnext/accounts/doctype/payment_reconciliation/test_payment_reconciliation.py

```diff
@@ -1845,6 +1846,107 @@ def test_cr_note_payment_limit_filter(self):
+
+		self.create_company()
+		self.create_cost_center()
+
+		# create bank account
+		parent_account = frappe.db.get_value(
+			"Account", {"company": self.company, "account_name": "Bank Accounts", "is_group": 1}, "name"
+		)
+		bank_account = create_account(
+			account_name="Bank Account",
+			account_type="Bank",
+			is_group=0,
+			company=self.company,
+			root_type="Asset",
+			report_type="Balance Sheet",
+			account_currency="INR",
+			parent_account=parent_account,
+			doctype="Account",
+		)
+
+		# create backdated fiscal year
+		create_fiscal_year(company=self.company, year_start_date="1990-04-01", year_end_date="1991-03-31")
+
+		# make journal entry for previous year
+		je_1 = frappe.new_doc("Journal Entry")
+		je_1.posting_date = "1990-06-01"
+		je_1.company = self.company
+		je_1.user_remark = "test"
+		je_1.set(
+			"accounts",
+			[
+				{
+					"account": self.debit_to,
+					"cost_center": self.cost_center,
+					"party_type": "Customer",
+					"party": self.customer,
+					"debit_in_account_currency": 0,
+					"credit_in_account_currency": 1000,
+				},
+				{
+					"account": bank_account,
+					"cost_center": self.sub_cc,
+					"credit_in_account_currency": 0,
+					"debit_in_account_currency": 500,
+				},
+				{
+					"account": "Cash - _PR",
+					"cost_center": self.sub_cc,
+					"credit_in_account_currency": 0,
+					"debit_in_account_currency": 500,
+				},
+			],
+		)
+		je_1.save()
+		je_1.submit()
+		je_1.reload()
+		# check journal entry is submitted
+		self.assertTrue(je_1.docstatus == 1)
+
+		# make period closing voucher
+		pcv = make_period_closing_voucher(
+			company=self.company, cost_center=self.cost_center, posting_date="1991-03-31"
+		)
+		pcv.reload()
+		# check if period closing voucher is completed
+		self.assertEqual(pcv.gle_processing_status, "Completed")
+
+		# make journal entry for active year
+		je_2 = self.create_journal_entry(
+			acc1=self.debit_to, acc2=self.income_account, amount=1000, posting_date=today()
+		)
+		je_2.accounts[0].party_type = "Customer"
+		je_2.accounts[0].party = self.customer
+		je_2.save()
+		je_2.submit()
+		je_2.reload()
+		# check journal entry is submitted
+		self.assertTrue(je_2.docstatus == 1)
+
+		# process reconciliation on closed period payment
+		pr = self.create_payment_reconciliation(party_is_customer=True)
+		pr.from_invoice_date = pr.to_invoice_date = pr.from_payment_date = pr.to_payment_date = None
+		pr.get_unreconciled_entries()
+		invoices = [invoice.as_dict() for invoice in pr.invoices]
+		payments = [payment.as_dict() for payment in pr.payments]
+		pr.allocate_entries(frappe._dict({"invoices": invoices, "payments": payments}))
+		pr.reconcile()
+		je_1.reload()
+		je_2.reload()
+
+		# check whether the payment reconciliation is done on the closed period
+		self.assertEqual(pr.get("invoices"), [])
+		self.assertEqual(pr.get("payments"), [])
+
+		# cancel created entires during test
+		pcv.cancel()
```

Comment:
No need to cancel / delete documents manually, as the DB rollback on tearDown() will take care of it.